### PR TITLE
[server] Reduce excessive logging in LeaderProducerCallback

### PR DESCRIPTION
**Pull request by sushantmane**
## Summary, imperative, start upper case, don't end with a period

Resolves #XXX

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: .github/workflows/gh-ci.yml [hunk 1]
```diff
@@ -231,7 +231,7 @@ jobs:
           rsync -R --files-from=artifacts.list . ${{ github.job }}-artifacts
           tar -zcvf ${{ github.job }}-artifacts.tar.gz ${{ github.job }}-artifacts
       - name: Upload Build Artifacts
-        if: success()
+        if: success() || failure()
```
Comment (ZacAttack) [inline .github/workflows/gh-ci.yml:234]: Is there a third state?  Why even have the if we'll do it on success or failure?

File: internal/venice-test-common/src/main/java/com/linkedin/venice/utils/InMemoryLogAppender.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package com.linkedin.venice.utils;
```
Comment (ZacAttack) [inline internal/venice-test-common/src/main/java/com/linkedin/venice/utils/InMemoryLogAppender.java:1]: Could we move this class into a test package?  This thing makes me a little nervous, wouldn't mind shuffling it away somewhere where it's unlikely to hurt anybody.

Comment (ZacAttack) [reply internal/venice-test-common/src/main/java/com/linkedin/venice/utils/InMemoryLogAppender.java:1]: Nevermind, I read the package name wrong.

**Review by ZacAttack — APPROVED** [review]
lgtm

File: .github/workflows/gh-ci.yml (hunk 1 above)
Comment (sushantmane) [reply .github/workflows/gh-ci.yml:234]: > Is there a third state? 

There is `always()` but it gets executed even if we cancel the job.

> Why even have the if we'll do it on success or failure?

If used with no condition or with `success()`, it collects logs only when a build is successful. 

With `failure()`  logs are collected only on build failure, not when the build is successful.

We want to collect logs on either build failure or success but not when a job is manually canceled.

**Comment by sushantmane** [comment]
Thanks for the review, @ZacAttack !

**Pull request merged**